### DISCUSSION TITLE
Supabase storage get list of files with size info

### DISCUSSION THREAD

**Discussion by wwawrzy** [supabase/supabase#5017] [Questions] [answered]
Hello,

I need to check some files from some of my storage bucket. Here is code:

 ```
StorageResponse<List<FileObject>> result =
        await supabase.storage.from(from).list(searchOptions: SearchOptions(limit: 10000, offset: 0, sortBy: SortBy(column: 'created_at', order: "desc")));
```

I obtain List<FileObject> properly. But I would like to know size without downloading it from supabase. It is possible?

Best regards

**Comment by GaryAustin1** [comment]
When I do list I get back a column called metadata that is jsonb and has the file size (at least for images).

**Reply by wwawrzy** [reply]
Thanks @GaryAustin1 for replay. In flutter (dekstop) I don't get any metadata at all.:

![image](https://user-images.githubusercontent.com/91059830/149726383-57f3805f-484e-4357-990c-efec0aff6164.png)

I assume you are probably on the mobile version.

**Comment by wwawrzy** [comment, marked as answer]
After a more careful check it turns out that in the supabase for flutter library (supabase: ^ 0.2.13) the metadata is in json.
```
     if (response.hasError) {
        return StorageResponse(error: response.error);
      } else {
        final fileObjects = List<FileObject>.from(
          (response.data as List).map(
            (item) => FileObject.fromJson(item),
          ),
        );
        return StorageResponse<List<FileObject>>(data: fileObjects);
```

However, when mapping to a FileObject list, they are skipped in Metadata.fromJson:

```
class Metadata {
  const Metadata({required this.name});

  Metadata.fromJson(dynamic json) : name = json['name'] as String?;

  final String? name;
}
```

**Reply by mzellner0** [reply]
Thanks for this discussion, it really helped me!
If anyone is looking for how to retrieve this information, you can create a stored procedure :

create or replace function select_metadata_photos_user(user_id uuid)
RETURNS TABLE (
  metadata jsonb
)
language plpgsql
as $$
begin
  return QUERY
    SELECT storage.objects.metadata from storage.objects where owner = user_id AND bucket_id = 'photos';
end;
$$;

For example, here I retrieve the metadata of photos belonging to a specific user.

Then, you can use it in your dart:
final response = await _client
      .rpc('select_metadata_photos_user', params: { 'user_id': userId })
      .execute();

**Reply by GaryAustin1** [reply]
@mzellner0 If this is running from client code you should just use auth.uid() in your function versus passing it.  This insures the only data you can retrieve is for the signed in user.   If you are calling it from a server, then passing the uuid is appropriate.

**Reply by mzellner0** [reply]
@GaryAustin1 I am calling this from my client side yes, so you're right, I will do that. Thanks for your help!
